cmd/cluster-agent/app/app.go

```diff
@@ -184,7 +185,9 @@ func start(cmd *cobra.Command, args []string) error {
 
 	log.Infof("Datadog Cluster Agent is now running.")
 
-	apiCl, err := apiserver.GetAPIClient() // make sure we can connect to the apiserver
+	retrierCtx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	apiCl, err := apiserver.WaitForAPIClient(retrierCtx) // make sure we can connect to the apiserver
 	if err != nil {
 		log.Errorf("Could not connect to the apiserver: %v", err)
```

Comment:
should we restart the DCA process instead of ignoring the error?

Reply:
5min can enough for a small network issue and it will definitely solve most of network glitch.
But I think if the DCA is not able to start properly the connection with the API server we should restart the process. and count on the container backoffpolicy to restart it severals time.

Reply:
you mean we should put an `os.Exit()` there ?

Reply:
I don't see any reason why the cluster agent should start if we don't have a working apiserver connection

Reply:
Maybe not `os.Exit()` directly but make sure the code returns and exit with error

Reply:
done